src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ChannelDispatcherBase.cs

```diff
@@ -10,10 +10,20 @@ public abstract class ChannelDispatcherBase : CommunicationObject
     {
         public abstract ServiceHostBase Host { get; }
 
+        internal void AttachInternal(ServiceHostBase host)
+        {
+            Attach(host);
+        }
+
         protected virtual void Attach(ServiceHostBase host)
         {
         }
 
+        internal void DetachInternal(ServiceHostBase host)
```

Comment:
This is code copied from .NET Framework. The whole ChannelDispatcherBase/ChannelDispatcher thing needs tidying up. I believe it's an artifact left over for a back compat scenario from an earlier release of WCF. Unfortunately ChannelDispatcherBase bleeds into ChannelDistpatcherCollection public api so people might be relying on the specific type in some extensibility. At some point there's going to be some compat breaks with modernizing some things like make message inspectors async. That's the time to clean this up. Until then, I'm keeping it the same as .NET Framework to prevent having to think about repercussions of changes too much.